backend/gene_info/api/ncbi_provider.py

```diff
@@ -0,0 +1,114 @@
+from backend.gene_info.config import GeneInfoConfig
+import logging
+import urllib.request
+import json
+import xml.etree.ElementTree as ET
+
+
+class NCBIException(Exception):
+    pass
+
+
+class NCBIAPIException(NCBIException):
+    pass
+
+
+class NCBIUnexpectedResultException(NCBIException):
+    pass
+
+
+class NCBIProvider(object):
+    """
+    Provider class used to generate NCBI URL
+    """
+
+    def __init__(self) -> None:
+        self.base_ncbi_uri = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/"
+        gene_info_config = GeneInfoConfig()
+        try:
+            self._api_key = f"&api_key={gene_info_config.ncbi_api_key}"
+        except RuntimeError:
+            logging.error("Could not find NCBI API key")
+            self._api_key = None
+
+    @property
```

Comment:
I dont think you actually need lines 34-40? setting it in the init function should be fine